termux-exec.c

```diff
@@ -230,9 +233,28 @@ int execve(const char* filename, char* const* argv, char* const* envp)
 			}
 		}
 	}
+	*filename_ptr = (char*)filename;
 
-	int ret = real_execve(filename, argv, envp);
 	free(new_argv);
 	free(new_envp);
```

Comment:
I'm afraid you'll have use-after-free bugs here? `argv` might point to `new_argv`, so if you call `real_exec*` after freeing the `new_argv`, you'll have a use-after-free issue here.

Maybe you should enable ASAN to test memory issues?